pkgs/applications/backup/nis-one-click-backup/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, cmake
+, ninja
+, qt6
+, wrapQtAppsHook
+, extra-cmake-modules
+}:
+
+stdenv.mkDerivation rec {
+  pname = "one-click-backup";
+  version = "1.2.2";
+
+  src = fetchFromGitLab {
+    owner = "dev-nis";
+    repo = "nis-one-click-backup-qt";
+    rev = "9a1ca839ca511d55716ffc8898e5d9ec9449ef18"; #version;
+    hash = "sha256-JbgmfKEpKRj/ph8dkW0IRulmAhk6Dwv/1Z4j3hUgPXk=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    wrapQtAppsHook
+    extra-cmake-modules
+  ];
+
+   buildInputs = [
+    qt6.full
```

Comment:
Please select only the parts of Qt that your package needs, `full` will pull alot of dependencies you very likely don't need.

Reply:
Well that's a problem I was already thinking about. The problem is, I'm not sure what parts my application needs. It is a QML application, but I didn't find a 'qt6.qml' or 'qt6.quick' or something like that on search.nixos.org (only qt6.qtquick3d, but that's not right, is it?), so I'm unsure what I should write here.

Reply:
Is qt6.qtbase enough for a QML application? Or is that only for Qt Widgets?

Reply:
For QML, try `qtdeclarative`.

![image](https://github.com/NixOS/nixpkgs/assets/23431373/da9a8b27-45f2-46dd-ba7c-d5795855e97d)

Reply:
I will try that, thanks!

Reply:
Should now be fixed with commit [64ae0e7](https://github.com/NixOS/nixpkgs/pull/287497/commits/64ae0e7aaf0da36b871fd875485fb8177437ec20)